staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1430,18 +1432,21 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 				klog.Warningf("Bad device-number %q for interface %s\n", numStr, macID)
 				continue
 			}
+			macDevNum[macID] = num
+		}
+
+		// Sort macIDs by interface device-number
+		sort.Slice(macIDs, func(i, j int) bool {
```

Comment:
i'm not sure under which case will trigger(or even will them trigger?)
`num, err := strconv.Atoi(strings.TrimSpace(numStr))` or `macID == ""` in above loop(which makes 'macDevNum[mac] == 0'
which will be sorted first as first mac, that's something we don't want :D

Reply:
Good call.  I've tweaked the loop a bit.